include/qore/QoreNamespace.h

```diff
@@ -309,4 +358,130 @@ class QoreNamespaceConstIterator {
     DLLEXPORT const QoreNamespace* get() const;
 };
 
+//! allows local namespaces to be iterated
+/** @since %Qore 0.8.13
+ */
+class QoreNamespaceNamespaceIterator {
```

Comment:
this class has implicit copy ctor and =operator, and the will do just a shallow copy and it will lead to undefined behavior, let's `=delete` them. Also let's `=default` the move ctor and =operator.